use-cases/inferencing/serving-with-vllm/README.md

```diff
@@ -0,0 +1,513 @@
+
+Cloud Monitoring provides an [importer](https://cloud.google.com/monitoring/dashboards/import-grafana-dashboards) that you can use to import dashboard files in the Grafana JSON format into Cloud Monitoring
+
+ Clone github repository
+
+  ```sh
+  git clone https://github.com/GoogleCloudPlatform/monitoring-dashboard-samples
+  ```
+
+ Change to the directory for the dashboard importer:
+
+  ```sh
+  cd monitoring-dashboard-samples/scripts/dashboard-importer
+  ```
+
+The dashboard importer includes the following scripts:
+
+- import.sh, which converts dashboards and optionally uploads the converted dashboards to Cloud Monitoring.
+- upload.sh, which uploads the converted dashboards—or any Monitoring dashboards—to Cloud Monitoring. The import.sh script calls this script to do the upload.
+
+ Import the dashboard
+
+  ```sh
+  ./import.sh ./configs/grafana.json ${PROJECT_ID}
+  ```
+
+  When you use the import.sh script, you must specify the location of the Grafana dashboards to convert. The importer creates a directory that contains the converted dashboards and other information.
+
+
+### Run Batch inference on GKE
+
+Once a model has completed fine-tuning and is deployed on GKE , its ready to run batch Inference pipeline.
+In this example batch inference pipeline, we would first send prompts to the hosted fine-tuned model and then validate the results based on ground truth.
+
+#### Prepare your environment
+
+
+Set env variables.
+
+```
+PROJECT_ID=<your-project-id>
+PROJECT_NUMBER=$(gcloud projects describe ${PROJECT_ID} --format="value(projectNumber)")
+CLUSTER_NAME=<your-gke-cluster>
+NAMESPACE=ml-serve
+MODEL_PATH=<your-model-path>
+BUCKET="<your dataset bucket name>"
+DATASET_OUTPUT_PATH=""
+ENDPOINT=<your-endpoint> # eg "http://vllm-openai:8000/v1/chat/completions"
+KSA=<k8s-service-account> # Service account with work-load identity enabled
+PREDICTIONS_FILE=<predictions.txt> # Look for sample example-prediction.text for expected results
+```
+
+Create Service account.
+
+```
+NAMESPACE=ml-serve
+kubectl create sa ${KSA} -n ${NAMESPACE}
+```
+
+Setup Workload Identity Federation access to read/write to the bucket for the inference batch data set
+
+```
+gcloud storage buckets add-iam-policy-binding gs://${BUCKET} \
+    --member "principal://iam.googleapis.com/projects/${PROJECT_NUMBER}/locations/global/workloadIdentityPools/${PROJECT_ID}.svc.id.goog/subject/ns/${NAMESPACE}/sa/${KSA}" \
+    --role "roles/storage.objectUser"
+```
+
+```
+gcloud storage buckets add-iam-policy-binding gs://${BUCKET} \
+    --member "principal://iam.googleapis.com/projects/${PROJECT_NUMBER}/locations/global/workloadIdentityPools/${PROJECT_ID}.svc.id.goog/subject/ns/${NAMESPACE}/sa/${KSA}" \
+    --role "roles/storage.legacyBucketWriter"
+```
+
+#### Build the image of the source and execute batch inference job
+
+Create Artifact Registry repository for your docker image
+
+```
+gcloud artifacts repositories create llm-inference-repository \
```

Comment:
we will have the default created registry by the platform

Reply:
We would need to add this one to the list to host inference artifacts in a separate folder/repository.

Reply:
thanks, will keep and note it as addition to the playground for ease-of-use